docs/sql-ref-literals.md

```diff
@@ -0,0 +1,378 @@
+---
+layout: global
+title: Literals
+displayTitle: Literals
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+A literal (also known as a constant) represents a fixed data value. Spark SQL supports the following literals:
+
+ * [String Literal](#string-literal)
+ * [Null Literal](#null-literal)
+ * [Boolean Literal](#boolean-literal)
+ * [Numeric Literal](#numeric-literal)
+ * [Datetime Literal](#datetime-literal)
+ * [Interval Literal](#interval-literal)
+
+### String Literal
+
+A string literal is used to specify a character string value.
+
+#### <em>Syntax</em>
```

Comment:
nit: We need `<em>`here to follow the other format?

Reply:
I think ```<em>``` makes the format a little prettier.

<img width="821" alt="Screen Shot 2020-04-17 at 6 36 45 PM" src="https://user-images.githubusercontent.com/13592258/79624979-7efda780-80da-11ea-86a4-ad3a4ad36600.png">


vs.

<img width="792" alt="Screen Shot 2020-04-17 at 6 36 00 PM" src="https://user-images.githubusercontent.com/13592258/79624983-8329c500-80da-11ea-80d2-cd696cf4bb96.png">

Reply:
I personally think the second one looks fine for the consistency with the others, e.g., 
https://github.com/apache/spark/blame/master/docs/sql-ref-syntax-qry-select-setops.md#L36

Reply:
OK I will remove <em>